### PR TITLE
GetCap: Add CAP_PCRS

### PR DESCRIPTION
**Pull request by AndreasFuchsSIT**
Add the capability for querying allocated PCRs

Signed-off-by: Andreas Fuchs <andreas.fuchs@sit.fraunhofer.de>

### CODE REVIEW

File: tools/tpm2_getcap.c [hunk 1]
```diff
@@ -849,6 +855,49 @@ dump_handles (TPM2_HANDLE     handles[],
...
+            case TPM2_ALG_SHA512:
+                tpm2_tool_output ("Bank %i (SHA512): ", i);
+                break;
+            default:
+                tpm2_tool_output ("Bank %i (ALG0x%08x): ", i,
+                                  pcrselection[i].hash);
+        }
+        first = true;
+        for (j = 0; j < 24; j++) {
+//            tpm2_tool_output ("\n\n%02x\n", pcrselection[i].pcrSelect[j / 8]);
```
Comment (williamcroberts) [inline tools/tpm2_getcap.c]: delete me

Comment (AndreasFuchsSIT) [reply tools/tpm2_getcap.c]: Of course

**Comment by AndreasFuchsSIT** [comment]
Shouldn't this already be tested via this ?
https://github.com/tpm2-software/tpm2-tools/blob/9597b55370b7050f867b92f941bb02ede4f5582b/test/integration/tests/getcap.sh#L64-L71

Looking into this however I saw that tools output yml ?

The current output of the cap is:
```
Bank 0 (SHA1): 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23
Bank 1 (SHA256): 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23
Bank 2 (SHA384): 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23
Bank 3 (SHA512): 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23
```

Shall I switch it to something more yamly ? Like
```
ActivePCRs: 
  - hash: sha1
    pcrs: [ 0, 1, 2 ]
  - hash: sha256
    pcrs: [ 0, 1, 2 ]
```

**Comment by AndreasFuchsSIT** [comment]
Ok, I just made it more yamly like this:
```yaml
$ ./tools/tpm2_getcap -Tmssim -c pcrs
activePCRs:
  - bank: sha1
    pcrs: [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23 ]
  - bank: sha256
    pcrs: [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23 ]
  - bank: sha384
    pcrs: [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23 ]
  - bank: sha512
    pcrs: [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23 ]
```

**Comment by williamcroberts** [comment]
I like when things are more yamaly.

> add a test.
> 
> I've been waiting for a patch like this, and I think it deprecates tpm2_nvlist.

N/M, thinking this was NV, its PCR.

**Pull request merged**